### ISSUE TITLE
Large amount of numerical errors in gradient

### ISSUE THREAD

**Issue by zgojcic** [vcg-uvic/learned-correspondence-release#11]
Hei, I am trying to train your network with the provided code but get a realy large number of numerical error in gradients. Is this normal? Even in the initial iterrations where the SVD based loss is not yet considered.

**Comment by tobiasploetz** [comment]
Hi,

we also observed numerical errors in the gradient. I think that the problem is related to the implementation of the classification loss, see [here](https://github.com/visinf/n3net/issues/5#issuecomment-466376865).

Bests,
Tobias

**Comment by zgojcic** [comment]
Hi, 
I can confirm that changing the loss function aleviates the errors and will report if the training breaks down. 

Thanks for your help, 

Regards,

Zan

**Comment by kmyid** [comment]
Hi, 

In our case, I don't recall this happening too often. Our method assumes that the input is in the calibrated camera coordinates, which typically is between -0.5 to 0.5. Could it be because of the data range? Anyways, please let us know!

Bests,
Kwang

**Comment by zgojcic** [comment]
Hi,
It is happening on the St. Peters data set and everything is done as you proposed. The fix by @tobiasploetz helped in our case and the training is not breaking down as can be seen in their examples. 

Best

Zan

**Comment by kmyid** [comment]
Hello, 

Thank you for letting us know!

Bests,
Kwang
